File: sicora-be-go/evalinservice/internal/domain/valueobjects/period_status.go
package valueobjects

// PeriodStatus representa el estado de un período de evaluación
type PeriodStatus string

const (
	PeriodStatusDraft  PeriodStatus = "DRAFT"  // Borrador, en preparación
	PeriodStatusActive PeriodStatus = "ACTIVE" // Activo, acepta evaluaciones
	PeriodStatusClosed PeriodStatus = "CLOSED" // Cerrado, no acepta más evaluaciones
)

// IsValid verifica si el estado del período es válido
func (ps PeriodStatus) IsValid() bool {
	switch ps {
	case PeriodStatusDraft, PeriodStatusActive, PeriodStatusClosed:
		return true
	default:
		return false
	}
}

// String retorna la representación en string del estado
func (ps PeriodStatus) String() string {
	return string(ps)
}

// CanAcceptEvaluations indica si el período puede aceptar evaluaciones
func (ps PeriodStatus) CanAcceptEvaluations() bool {
	return ps == PeriodStatusActive
}

// CanBeModified indica si el período puede ser modificado
func (ps PeriodStatus) CanBeModified() bool {
	return ps == PeriodStatusDraft
}

// CanBeActivated indica si el período puede ser activado
func (ps PeriodStatus) CanBeActivated() bool {
	return ps == PeriodStatusDraft
}

// CanBeClosed indica si el período puede ser cerrado
func (ps PeriodStatus) CanBeClosed() bool {
	return ps == PeriodStatusActive
}

File: sicora-be-python/evalinservice/app/domain/value_objects/period_status.py
"""Period status value object."""

from enum import Enum


class PeriodStatus(Enum):
    """
    Value object que define los estados posibles de un período de evaluación.
    """
    SCHEDULED = "scheduled"  # Programado para el futuro
    ACTIVE = "active"  # Activo, evaluaciones pueden ser realizadas
    INACTIVE = "inactive"  # Inactivo temporalmente
    CLOSED = "closed"  # Cerrado, no se pueden realizar más evaluaciones
    EXPIRED = "expired"  # Expirado por fecha
    
    @classmethod
    def get_display_name(cls, status: "PeriodStatus") -> str:
        """Retorna el nombre para mostrar del estado."""
        display_names = {
            cls.SCHEDULED: "Programado",
            cls.ACTIVE: "Activo",
            cls.INACTIVE: "Inactivo",
            cls.CLOSED: "Cerrado",
            cls.EXPIRED: "Expirado"
        }
        return display_names.get(status, status.value)
    
    @classmethod
    def get_description(cls, status: "PeriodStatus") -> str:
        """Retorna una descripción del estado."""
        descriptions = {
            cls.SCHEDULED: "El período está programado para iniciar en el futuro",
            cls.ACTIVE: "El período está activo y se pueden realizar evaluaciones",
            cls.INACTIVE: "El período está temporalmente inactivo",
            cls.CLOSED: "El período ha sido cerrado manualmente",
            cls.EXPIRED: "El período ha expirado por fecha"
        }
        return descriptions.get(status, "Estado desconocido")
    
    @classmethod
    def allows_evaluations(cls, status: "PeriodStatus") -> bool:
        """Verifica si el estado permite realizar evaluaciones."""
        return status == cls.ACTIVE
    
    @classmethod
    def can_be_modified(cls, status: "PeriodStatus") -> bool:
        """Verifica si el período puede ser modificado en este estado."""
        return status in [cls.SCHEDULED, cls.INACTIVE]
    
    @classmethod
    def is_final_state(cls, status: "PeriodStatus") -> bool:
        """Verifica si el estado es final (no se puede cambiar)."""
        return status in [cls.CLOSED, cls.EXPIRED]
